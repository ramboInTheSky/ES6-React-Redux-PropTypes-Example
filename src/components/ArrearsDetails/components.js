import styled from 'styled-components';
import { Button, Typography, mediaQueries } from 'nhh-styles';

export const Heading = styled(Typography.H3)`
  margin-bottom: 1.5rem;
`;

export const Label = styled.span`
  font-weight: 700;
  ${mediaQueries.desktop`
    white-space: nowrap;
  `};
  ${mediaQueries.tablet`
    white-space: nowrap;
  `};
`;

export const LabelLink = styled(Button)`
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
`;

export const Wrapper = styled.div`
  margin-bottom: 1.5rem;
`;
