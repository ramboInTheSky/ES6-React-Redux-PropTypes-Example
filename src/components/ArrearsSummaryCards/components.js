import styled from 'styled-components';
import { mediaQueries } from 'nhh-styles';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 12px;

  > a {
    color: inherit;
    margin-bottom: 16px;
    margin-right: 1%;
    width: 32%;

    &:hover {
      color: inherit;
      text-decoration: none;
    }

    ${mediaQueries.mobile`
      margin-right: 0;
      width: 100%;
    `};

    ${mediaQueries.tablet`
      width: 49%;
    `};
  }
`;
