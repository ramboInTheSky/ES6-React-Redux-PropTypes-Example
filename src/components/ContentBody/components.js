import styled from 'styled-components';

export const OuterContainer = styled.div`
  padding-top: 50px;
  padding-bottom: 50px;
`;
export const InnerContainer = styled.div``;
