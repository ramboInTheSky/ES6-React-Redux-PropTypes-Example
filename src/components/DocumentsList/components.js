import styled from 'styled-components';
import { Button, mediaQueries, Typography } from 'nhh-styles';

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;

export const Link = styled(Button)`
  font-size: inherit;
  font-weight: 400;
  display: inline;
`;

export const Container = styled.section`
  padding: 25px 28px 0;

  ${mediaQueries.mobile`
    padding: 15px 18px;
  `};
`;

export const ErrorMessage = styled.div`
  color: ${({ theme }) => (theme ? theme.colors.support.two : 'inherit')};
  margin-bottom: 50px;
`;

export const DateField = styled.div``;

export const Row = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.colors.tertiaryLighter};
  color: ${({ theme }) => theme.colors.tertiary};
  display: flex;
  flex-direction: row;
  font-size: 16px;
  padding: 8px 0;
`;

export const Header = Row.extend`
  font-size: 14px;
  font-weight: bold;
  text-transform: uppercase;
`;

export const Column = styled.div`
  width: ${({ width }) => width || 'auto'};
  text-align: ${({ align }) => align || 'left'};
  padding-right: 8px;
  word-wrap: break-word;

  &:last-child {
    padding-right: 0;
  }

  ${mediaQueries.mobile`
    padding-right: 4px;
  `};
`;

export const DesktopView = styled.div`
  display: none;

  ${mediaQueries.desktop`
    display: block;
  `};
`;

export const MobileAndTabletView = styled.div`
  display: block;

  ${mediaQueries.desktop`
    display: none;
  `};
`;
