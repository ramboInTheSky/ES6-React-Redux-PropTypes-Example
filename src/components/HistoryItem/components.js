import styled from 'styled-components';
import { Typography } from 'nhh-styles';

export const Label = styled.span`
  font-weight: 700;
`;

export const Title = styled(Typography.H3)`
  margin: -30px 0 30px 0;
`;

export const Value = styled.span`
  word-break: break-word;
`;
