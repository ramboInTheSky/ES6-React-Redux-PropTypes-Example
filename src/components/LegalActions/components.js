import styled, { withTheme } from 'styled-components';
import { Typography } from 'nhh-styles';

export const Wrapper = styled.div`
  margin: 1.5rem 0 1.875rem 0;
`;

export const P = styled(Typography.P)`
  margin: 1rem 0;
`;

export const Label = styled.span`
  font-weight: 700;
`;

export const Highlight = withTheme(styled.span`
  color: ${({ theme, isExpiring }) => (isExpiring ? theme.colors.support.two : 'inherit')};
  ${({ isExpiring }) => isExpiring && `font-weight: 700;`};
`);
