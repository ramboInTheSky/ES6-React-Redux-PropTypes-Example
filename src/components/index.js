export { default as ArrearsDetails } from './ArrearsDetails';
export { default as ArrearsSummaryCards } from './ArrearsSummaryCards';
export { default as ButtonRow } from './ButtonRow';
export { default as ContentBody } from './ContentBody';
export { default as ErrorBoundary } from './ErrorBoundary';
export { default as FieldRow } from './FieldRow';
export { default as FormWrapper } from './FormWrapper';
export { default as Heading } from './Heading';
export { default as HistoryItem } from './HistoryItem';
export { default as LegalActions } from './LegalActions';
export { default as MediaTableView } from './MediaTableView';
export { default as ModalFieldRow } from './ModalFieldRow';
export { default as NoTouching } from './NoTouching';
export { default as PreviewLinkContainer } from './PreviewLinkContainer';
export { default as PreviewTemplate } from './PreviewTemplate';
export { default as PreviewLink } from './PreviewLink';
export { default as NotificationWrapper } from './NotificationWrapper';
export { default as TaskTitle } from './TaskTitle';
export { default as TasksSummaryCards } from './TasksSummaryCards';
export { default as TemplateContainer } from './TemplateContainer';
