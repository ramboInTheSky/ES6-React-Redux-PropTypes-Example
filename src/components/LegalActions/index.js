import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import format from 'string-format';
import path from 'ramda/src/path';
import { formatting, PanelListItem, Typography } from 'nhh-styles';

import { Highlight, Label, P, Wrapper } from './components';

const renderLegalAction = ({ expiringSoon, expiryDate, labels }) => (action, bdd) => (
  <PanelListItem
    alert={action.notifyOfPendingExpiry}
    dataBdd={bdd}
    title={action.title}
    href={path(['referralPack', 'legalReferralPack', 'link'], action)}
  >
    <P>
      {!!action.expiryDate && (
        <Fragment>
          <Highlight isExpiring={!!action.notifyOfPendingExpiry}>
            <Label>{labels.expires}</Label>{' '}
            <span data-bdd={`${bdd}-expiry-date`}>
              {format(expiryDate, {
                date: formatting.formatDate(action.expiryDate),
                alert: action.notifyOfPendingExpiry ? expiringSoon : '',
              })}
            </span>
          </Highlight>
          <br />
        </Fragment>
      )}
      {!!action.createdOn && (
        <Fragment>
          <Label>{labels.createdOn}</Label>{' '}
          <span data-bdd={`${bdd}-created-on`}>{formatting.formatDate(action.createdOn)}</span>
          <br />
        </Fragment>
      )}
      {!!action.status && (
        <Fragment>
          <Label>{labels.status}</Label> <span data-bdd={`${bdd}-status`}>{action.status}</span>
          <br />
        </Fragment>
      )}
      {!!path(['raisedBy', 'name'], action) && (
        <Fragment>
          <Label>{labels.owner}</Label>{' '}
          <span data-bdd={`${bdd}-owner`}>{path(['raisedBy', 'name'], action)}</span>
          <br />
        </Fragment>
      )}
    </P>
  </PanelListItem>
);

const LegalActions = ({ heading, legalReferral, primaryLegalAction, ...props }) => {
  const renderAction = renderLegalAction(props);
  return (
    <Fragment>
      <Typography.H3>{heading}</Typography.H3>
      <Wrapper>
        {!!primaryLegalAction && renderAction(primaryLegalAction, 'primaryLegalAction')}
        {!!legalReferral && renderAction(legalReferral, 'legalReferral')}
      </Wrapper>
    </Fragment>
  );
};

export const legalActionsPropTypes = {
  expiringSoon: PropTypes.string.isRequired,
  expiryDate: PropTypes.string.isRequired,
  heading: PropTypes.string.isRequired,
  labels: PropTypes.shape({
    createdOn: PropTypes.string.isRequired,
    expires: PropTypes.string.isRequired,
    owner: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
  }).isRequired,
};

export const referralPackPropTypes = {
  legalReferralPack: PropTypes.shape({
    link: PropTypes.string.isRequired,
  }).isRequired,
};

export const legalActionPropTypes = {
  createdOn: PropTypes.string.isRequired,
  expiryDate: PropTypes.string,
  headingExpiring: PropTypes.string,
  notifyOfPendingExpiry: PropTypes.bool,
  raisedBy: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
  referralPack: PropTypes.shape(referralPackPropTypes),
  title: PropTypes.string.isRequired,
};

LegalActions.defaultProps = {
  legalReferral: null,
  primaryLegalAction: null,
};

LegalActions.propTypes = {
  ...legalActionsPropTypes,
  legalReferral: PropTypes.shape(legalActionPropTypes),
  primaryLegalAction: PropTypes.shape(legalActionPropTypes),
};

export default LegalActions;
