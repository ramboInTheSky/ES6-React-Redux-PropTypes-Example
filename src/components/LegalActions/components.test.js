import React from 'react';
import { shallow } from 'enzyme';

import 'jest-styled-components';

import { testUtils } from 'nhh-styles';

import { Wrapper, P, Label, Highlight } from './components';

describe('LegalActions components', () => {
  const theme = {
    colors: {
      support: {
        two: 'red',
      },
    },
  };

  it('should render <Wrapper /> correctly', () => {
    expect(shallow(<Wrapper />)).toMatchSnapshot();
  });
  it('should render <P /> correctly', () => {
    expect(shallow(<P />)).toMatchSnapshot();
  });
  it('should render <Label /> correctly', () => {
    expect(shallow(<Label />)).toMatchSnapshot();
  });
  it('should render <Highlight /> correctly', () => {
    expect(shallow(<Highlight theme={theme} />)).toMatchSnapshot();
  });
  it('should render <Highlight /> correctly with isExpiring - true', () => {
    const wrapper = testUtils.renderWithTheme(<Highlight theme={theme} isExpiring />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render <Highlight /> correctly with isExpiring - false', () => {
    const wrapper = testUtils.renderWithTheme(<Highlight theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
