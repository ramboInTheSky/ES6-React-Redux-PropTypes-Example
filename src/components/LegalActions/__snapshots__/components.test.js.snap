// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegalActions components should render <Highlight /> correctly 1`] = `
<styled.span
  theme={
    Object {
      "colors": Object {
        "support": Object {
          "two": "red",
        },
      },
    }
  }
/>
`;

exports[`LegalActions components should render <Highlight /> correctly with isExpiring - false 1`] = `
.c0 {
  color: inherit;
}

<span
  className="c0"
/>
`;

exports[`LegalActions components should render <Highlight /> correctly with isExpiring - true 1`] = `
.c0 {
  color: red;
  font-weight: 700;
}

<span
  className="c0"
/>
`;

exports[`LegalActions components should render <Label /> correctly 1`] = `
.c0 {
  font-weight: 700;
}

<span
  className="c0"
/>
`;

exports[`LegalActions components should render <P /> correctly 1`] = `
.c0 {
  margin: 1rem 0;
}

<styled.p
  className="c0"
/>
`;

exports[`LegalActions components should render <Wrapper /> correctly 1`] = `
.c0 {
  margin: 1.5rem 0 1.875rem 0;
}

<div
  className="c0"
/>
`;
